CDCgov/prime-reportstream
# 15314: use correct filename logic for new deliveries API

This PR updates the new delivery API to set the filename to be what the receiver would have gotten

Test Steps:
1. Run the deliveries API
```sh
curl --location 'http://localhost:7071/api/v1/waters/org/ignore/deliveries?receivingOrgSvcStatus=ACTIVE%2CTESTING' \
--header 'authentication-type: okta' \
--header 'Content-Type: application/json' \
--header 'Authorization: Bearer dummy' \
--data '{
    "sort": {
        "direction": "DESC",
        "property": "createdAt"
    },
    "pagination": {
        "page": 1,
        "limit": 10
    },
    "filters": []
}'
```
2. Grab one the ids
3. Run the download report API
```sh
curl --location 'http://localhost:7071/api/history/report/{ID}' \
--header 'organization: ignore' \
--header 'Authorization: Bearer dummy'
```
4. Confirm the filename is the same

## Changes
- After fetching the rows from the db, processes the DeliveryHistory models to correctly set the filename

## Checklist

### Testing

### Process

## Linked Issues
- https://github.com/CDCgov/prime-reportstream/issues/15314

## To Be Done

## Specific Security-related subjects a reviewer should pay specific attention to
